packages/shared/createRef/index.ts

```diff
@@ -0,0 +1,22 @@
+import type { Ref, ShallowRef } from 'vue'
+import { ref as deepRef, shallowRef } from 'vue'
+
+export type MaybeDeepRef<T = any, D extends boolean = false> = D extends true ? Ref<T> : ShallowRef<T>
+
+export function createRef<T = any, D extends boolean = false>(value: T, deep?: D): MaybeDeepRef<T, D> {
```

Comment:
maybe better naming here 😅

Reply:
The name sounds good to me. We should add some jsdoc